src/api/resolvedTimeline.ts

```diff
@@ -0,0 +1,135 @@
+ * needs to be re-resolved afterwards.
+ */
+export interface ResolvedTimeline<TContent extends Content = Content> {
+	/** The options used to resolve the timeline */
+	// options: ResolveOptions
+	/** Map of all objects on timeline */
+	objects: ResolvedTimelineObjects<TContent>
+	/** Map of all classes on timeline, maps className to object ids */
+	classes: { [className: string]: Array<string> }
+	/** Map of the object ids, per layer */
+	layers: { [layer: string]: Array<string> }
+
+	// states: AllStates
+	nextEvents: Array<NextEvent>
+
+	statistics: {
+		/** Number of objects that were unable to resolve */
+		unresolvedCount: number
+		/** Number of objects that were resolved */
+		resolvedCount: number
+
+		/** Number of resolved instances */
+		resolvedInstanceCount: number
+		/** Number of resolved objects */
+		resolvedObjectCount: number
+		/** Number of resolved groups */
+		resolvedGroupCount: number
+		/** Number of resolved keyframes */
+		resolvedKeyframeCount: number
+
+		/** How many objects that was actually resolved (is affected when using cache) */
+		resolvingCount: number
+	}
+}
+export interface ResolvedTimelineObjects<TContent extends Content = Content> {
+	[id: string]: ResolvedTimelineObject<TContent>
+}
+export interface ResolvedTimelineObject<TContent extends Content = Content> extends TimelineObject<TContent> {
+	resolved: {
+		/** Instances of the object on the timeline */
+		instances: Array<TimelineObjectInstance>
+		/** A number that increases the more levels inside of a group the objects is. 0 = no parent */
+		levelDeep: number
+		/** Id of the parent object (for children in groups or keyframes) */
+		parentId: string | undefined
+		/** True if object is a keyframe */
+		isKeyframe: boolean
+
+		/** Is set to true while object is resolved (to prevent circular references) */
+		resolving: boolean
+
+		/** Is set to true when object is resolved first time, and isn't reset thereafter */
+		firstResolved: boolean
+
+		/** Is set to true when object's references has been resolved */
+		resolvedReferences: boolean
+		/** Is set to true when object's conflicts has been resolved */
+		resolvedConflicts: boolean
+
+		/** True if object is referencing itself (only directly, not indirectly via another object) */
+		isSelfReferencing: boolean
+		/** Ids of all other objects that directly affects this object (ie through direct reference, classes, etc) */
+		directReferences: Reference[]
+	}
+}
+export interface InstanceBase {
+	/** The start time of the instance */
+	start: Time
+	/** The end time of the instance (null = infinite) */
+	end: Time | null
+}
+export interface TimelineObjectInstance extends InstanceBase {
+	/** id of the instance (unique)  */
+	id: InstanceId
+	/** if true, the instance starts from the beginning of time */
+	isFirst?: boolean
+
+	/** The original start time of the instance (if an instance is split or capped, the original start time is retained in here).
```

Comment:
I'm new to this, so could you clarify if I got it right: Is this the original start time of the _object_ that was turned into one or more instances?

Reply:
Almost, but not really.
Lets say an object A has `start: #obj.start, duration: 50` which gives it the instance: `10-50`.

Now there's another object B on that same layer, at `20-30`.

 B interrupts A, so A gets the instances `[{start: 10, end: 20}, {start: 30, end: 50, originalStart: 10}] `.

So originslStart refers to the original instance when an instance os affected/split by a conflict. We use this for example om TSR where we want a video two continue playing (and not restart from the beginning) when it is interrupted.